clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -192,10 +192,17 @@ public static Errors forCode(short code) {
     }
 
     /**
-     * Return the error instance associated with this exception (or UNKNOWN if there is none)
+     * Return the error instance associated with this exception or any of its superclasses (or UNKNOWN if there is none)
      */
     public static Errors forException(Throwable t) {
-        Errors error = classToError.get(t.getClass());
+        Errors error = null;
+        Class clazz = t.getClass();
+        while (clazz != null) {
+            error = classToError.get(clazz);
+            if(error != null)
+                break;
```

Comment:
I'd probably just `return error` here (I'm not a fan of `return`, but `break` isn't much better and I assume you wrote it like this to avoid a call to `getSuperclass()` in the very common case where we find the `Errors` on the first attempt). While I am nitpicking: space after `if` is missing.